assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -214,21 +214,21 @@ const baseActions = {
 	 * Resets the publications data in the store.
 	 *
 	 * @since n.e.x.t
-	 *
-	 * @return {Object} The dispatched action results.
 	 */
 	*resetPublications() {
 		const registry = yield commonActions.getRegistry();
 
+		yield errorStoreActions.clearErrors( 'getPublications' );
+
+		yield commonActions.await(
+			registry
+				.dispatch( MODULES_READER_REVENUE_MANAGER )
+				.invalidateResolutionForStoreSelector( 'getPublications' )
+		);
```

Comment:
I would think the resolver should be invalidated after the state is reset below, otherwise it may run and not fetch if publications are set in state. Is there a reason the order has changed here?

Reply:
Ah, sorry, I had mentioned this in [the original PR](https://github.com/google/site-kit-wp/pull/9167#discussion_r1707033246). The reason I changed the order here is because when the publications were cleared in state before invalidating resolution, our `useEffect` [here](https://github.com/google/site-kit-wp/blob/62dabdac109b1801851657ad50c463821b6643b2/assets/js/modules/reader-revenue-manager/components/setup/SetupMain.js#L83-L99) ran and caused the `PublicationCreate` component to appear, which we don't want.

According to my tests, this re-order seems to work well. Do you have any other ideas that may work here?

Reply:
If it works and tests are happy that works for me. I'm not sure we should be using the form state like that but it's fine for now 👍